### PR TITLE
Feat(frontend): improve the soft delete empty state 

### LINKED ISSUE #6834: Customize soft-delete empty state
# Current Behavior

We display the "No records" empty state, which is confusing. Additionally, we offer the ability to add a record from the soft delete, which should not be allowed.

CleanShot 2024-09-02 at 11 22 55

# Desired Behavior

Display the following empty state: The "Remove Deleted Filter" should remove the deleted filter to allow the user to see non-deleted records.

image

https://www.figma.com/design/xt8O9mFeLl46C5InWwoMrN/Twenty?node-id=38833-78611&node-type=FRAME&t=W5U3Vvrsy9T8wUjD-11

### PR DESCRIPTION
**Pull request by pacyL2K19**
# This PR

- Fix #6834 

## Demo

https://www.loom.com/share/235c4425f3264f429e2064a9d1604a90?sid=02a815c9-3b1a-45e6-b5ce-d5eb3b40e10e

## Notes

- There is a missing icon in Figma corresponding to the `noDeletedRecordFound` in the dark mode, thus I used the same icon (different background because we have the correct background image) for both dark / light modes
Screenshot 2024-09-03 at 15 04 57
cc: @Bonapara

### CODE REVIEW

**Comment by Bonapara** [comment]
CleanShot 2024-09-03 at 15 12 24

Added the dark mode for you @pacyL2K19 ;)

File: packages/twenty-front/src/modules/object-record/record-index/hooks/useHandleToggleTrashColumnFilter.ts [hunk 1]
```diff
@@ -63,5 +66,12 @@ export const useHandleToggleTrashColumnFilter = ({
     upsertCombinedViewFilter(newFilter);
   }, [columnDefinitions, objectMetadataItem, upsertCombinedViewFilter]);
 
-  return handleToggleTrashColumnFilter;
+  const toggleSoftDeleteFilterState = useRecoilCallback(
+    ({ set }) =>
+      (currentState?: boolean) => {
+        set(isSoftDeleteActiveState, Boolean(currentState));
```
Comment (lucasbordeau) [inline packages/twenty-front/src/modules/object-record/record-index/hooks/useHandleToggleTrashColumnFilter.ts:72]: Why wrapping in Boolean ?

File: packages/twenty-front/src/modules/object-record/record-index/hooks/useHandleToggleTrashColumnFilter.ts [hunk 2]
```diff
@@ -63,5 +66,12 @@ export const useHandleToggleTrashColumnFilter = ({
...
+      },
+    [isSoftDeleteActiveState],
+  );
+  return [handleToggleTrashColumnFilter, toggleSoftDeleteFilterState];
```
Comment (lucasbordeau) [inline packages/twenty-front/src/modules/object-record/record-index/hooks/useHandleToggleTrashColumnFilter.ts:76]: Use named export please : 

{
  handleToggleTrashColumnFilter,
  toggleSoftDeleteFilterState
}

File: packages/twenty-front/src/modules/object-record/record-table/components/RecordTableEmptyState.tsx [hunk 3]
```diff
@@ -10,24 +12,51 @@ import {
...
+      tableFilters.find(
+        (filter) =>
+          filter.definition.label === 'Deleted at' &&
+          filter.operand === 'isNotEmpty',
+      )?.id ?? '',
+    );
+    toggleSoftDeleteFilterState(false);
+  };
+
   const [title, subTitle, Icon, onClick, buttonTitle] = isRemote
```
Comment (lucasbordeau) [inline packages/twenty-front/src/modules/object-record/record-table/components/RecordTableEmptyState.tsx:60]: Please refactor this part by creating one variable for each variable, so it's more readable in a sequential order.

**Review by lucasbordeau — COMMENTED** [review]
Nice work thank you ! Just some minor comments.

File: packages/twenty-front/src/modules/object-record/record-index/hooks/useHandleToggleTrashColumnFilter.ts (hunk 1 above)
Comment (pacyL2K19) [reply packages/twenty-front/src/modules/object-record/record-index/hooks/useHandleToggleTrashColumnFilter.ts:72]: Fixed this as well

**Review by lucasbordeau — APPROVED** [review]

**Pull request merged**